src/App/Fossa/Analyze.hs

```diff
@@ -411,8 +411,11 @@ analyze cfg = Diag.context "fossa-analyze" $ do
   -- If we find nothing but keyword search, we exit with an error, but explain that the error may be ignorable.
   -- We do not want to succeed, because nothing gets uploaded to the API for keyword searches, so `fossa test` will fail.
   -- So the solution is to still fail, but give a hopefully useful explanation that the error can be ignored if all you were expecting is keyword search results.
+  -- In the case that we don't find any analysis targets, we want to emit a warning and upload empty source units.
   case (keywordSearchResultsFound, checkForEmptyUpload includeAll projectScans filteredProjects' additionalSourceUnits licenseSourceUnits) of
-    (False, NoneDiscovered) -> Diag.fatal ErrNoProjectsDiscovered
+    (False, NoneDiscovered) -> do
+      Diag.warn ErrNoProjectsDiscovered
+      doUpload outputResult iatAssertion destination basedir jsonOutput revision $ SourceUnitOnly []
     (True, NoneDiscovered) -> Diag.fatal ErrOnlyKeywordSearchResultsFound
     (False, FilteredAll) -> Diag.fatal ErrFilteredAllProjects
     (True, FilteredAll) -> Diag.fatal ErrOnlyKeywordSearchResultsFound
```

Comment:
Looks great, we should apply this to all the other cases too.
Probably this looks like:
1. Change the `case` branches to just emit warnings, and evaluate to the `scanUnits` to upload
2. After the `case` statement, do the upload with the returned `scanUnits`